src/Generator/Generators/CSharp/CSharpMarshal.cs

```diff
@@ -492,7 +492,7 @@ public override bool VisitPointerType(PointerType pointer, TypeQualifiers quals)
 
             var pointee = pointer.Pointee.Desugar();
             if (Context.Function != null && pointer.IsPrimitiveTypeConvertibleToRef() &&
-                Context.MarshalKind != MarshalKind.VTableReturnValue)
+                Context.MarshalKind != MarshalKind.VTableReturnValue && Context.Function.OperatorKind != CXXOperatorKind.Subscript)
```

Comment:
Move the new condition to a new line - notice that this line is too long compared to the others.